django_kmatch/fields.py

```diff
@@ -27,10 +29,19 @@ def get_db_prep_value(self, value, connection, prepared=False):
         """
         if isinstance(value, K):
             value = value.pattern
+
+        if not self.null and value is None:
+            value = json.dumps(value)
+
         return super(KField, self).get_db_prep_value(value, connection, prepared=False)
 
     def to_python(self, value):
         """
         Used to obtain a K object for a provided pattern.
         """
-        return K(value) if not isinstance(value, K) and value is not None else value
+        if not isinstance(value, K) and value is not None:
+            if isinstance(value, str):
+                return json.loads(value)
```

Comment:
i don't think this logic is the same. if the value is a string, we still want to return a K object containing the parsed pattern right?

Reply:
if the purpose of this method is to turn the pattern into a K object, i would expect this to return K(json.loads(value)). the other two returns in this method are returning either K object or null.